### PR TITLE
TF-3113 Fix cannot empty spam on mobile

### PR DESCRIPTION
**Pull request by dab246**
## Issue

#3113 

## Reproduced

[reproduced.webm](https://github.com/user-attachments/assets/41525a2b-c933-4bcb-9b5b-6a2d16f81e9d)

## Root cause

- [SendPort.send()](https://api.flutter.dev/flutter/dart-isolate/SendPort/send.html) of isolate flutter only supports some objects
- Network connection objects cannot be shared between isolates.

Screenshot 2024-09-04 at 10 41 49

## Resolved

[demo-spam.webm](https://github.com/user-attachments/assets/c82868e1-95ea-4c62-b7e9-70cb8e71c685)

### CODE REVIEW

**Review by tddang-linagora — APPROVED** [review]

File: lib/features/thread/data/network/thread_isolate_worker.dart [hunk 1]
```diff
@@ -66,12 +66,13 @@ class ThreadIsolateWorker {
...
     TypeSendPort sendPort
   ) async {
-    final rootIsolateToken = args.isolateToken;
-    BackgroundIsolateBinaryMessenger.ensureInitialized(rootIsolateToken);
-    await HiveCacheConfig.instance.setUp();
-
-    List<EmailId> emailListCompleted = List.empty(growable: true);
     try {
+      final rootIsolateToken = args.isolateToken;
+      BackgroundIsolateBinaryMessenger.ensureInitialized(rootIsolateToken);
```
Comment (hoangdat) [inline lib/features/thread/data/network/thread_isolate_worker.dart:71]: what is this purpose?

File: lib/main/bindings/network/network_isolate_binding.dart [hunk 2]
```diff
@@ -24,41 +29,57 @@ class NetworkIsolateBindings extends Bindings {
...
+    Get.put(DioClient(
+      Get.find<Dio>(tag: BindingTag.isolateTag)),
+      tag: BindingTag.isolateTag);
     Get.put(const FlutterAppAuth(), tag: BindingTag.isolateTag);
     Get.put(AppAuthWebPlugin(), tag: BindingTag.isolateTag);
-    Get.put(AuthenticationClientBase(tag: BindingTag.isolateTag), tag: BindingTag.isolateTag);
-    _bindingInterceptors(dio);
+    Get.put(AuthenticationClientBase(
+      tag: BindingTag.isolateTag),
+      tag: BindingTag.isolateTag);
```
Comment (hoangdat) [inline lib/main/bindings/network/network_isolate_binding.dart:47]:
```suggestion
    Get.put(
      AuthenticationClientBase(tag: BindingTag.isolateTag),
      tag: BindingTag.isolateTag
    );
```

File: lib/main/bindings/network/network_isolate_binding.dart [hunk 3]
```diff
@@ -78,4 +99,19 @@ class NetworkIsolateBindings extends Bindings {
...
+  void _bindingSharing() {
+    Get.put(OIDCHttpClient(
+      Get.find<DioClient>(tag: BindingTag.isolateTag)),
+      tag: BindingTag.isolateTag);
+    Get.put(IOSSharingManager(
+      Get.find<KeychainSharingManager>(),
+      Get.find<StateCacheManager>(),
+      Get.find<TokenOidcCacheManager>(),
+      Get.find<AuthenticationInfoCacheManager>(),
+      Get.find<OidcConfigurationCacheManager>(),
```
Comment (hoangdat) [inline lib/main/bindings/network/network_isolate_binding.dart:112]: Why no need to bind these with tag?

File: lib/features/thread/data/network/thread_isolate_worker.dart (hunk 1 above)
Comment (dab246) [reply lib/features/thread/data/network/thread_isolate_worker.dart:71]: From flutter 3.7 to use plugins and platform channels from any isolate we need BackgroundIsolateBinaryMessenger. This is not a new addition, please read more from previous issues.

File: lib/main/bindings/network/network_isolate_binding.dart (hunk 3 above)
Comment (dab246) [reply lib/main/bindings/network/network_isolate_binding.dart:112]: No need for it to be objects from the database, we always reset it every time we instantiate a new isolate.

File: lib/main/bindings/network/network_isolate_binding.dart (hunk 2 above)
Comment (dab246) [reply lib/main/bindings/network/network_isolate_binding.dart]: done

**Review by hoangdat — APPROVED** [review]

**Pull request merged**